src/Microsoft.Azure.WebJobs.Host/Singleton/IDistributedLockManager.cs

```diff
@@ -0,0 +1,76 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Host.Executors;
+
+namespace Microsoft.Azure.WebJobs.Host
+{
+    /// <summary>
+    /// Manage distributed lock. A lock is specified by (account, lockId).  This implementation should cooperate with <see cref="SingletonConfiguration"/>
+    /// </summary>
+    /// <remarks>
+    /// The default implementation of this is based on blob leases. 
+    /// 1. Account can be null or it may be a storage account name intended for <see cref="IStorageAccountProvider"/>. 
+    /// 2. lockId has the same naming restrictions as blobs.     
+    /// </remarks>    
+    public interface IDistributedLockManager
+    {        
+        /// <summary>
+        /// Try to acquire a lock specified by (account, lockId).                 
+        /// </summary>
+        /// <param name="account">optional. A string specifying the account to use. LockIds are scoped to an account </param>
+        /// <param name="lockId">the name of the lock. </param>
+        /// <param name="lockOwnerId">a string hint specifying who owns this lock. Only used for diagnostics. </param>
+        /// <param name="proposedLeaseId">optional. This can allow the caller to immediately assume the lease.</param>
+        /// <param name="lockPeriod">The length of the lease to acquire. The caller is responsible for Renewing the lease well before this time is up. 
+        /// The exact value here is restricted based on the underlying implementation.  </param>
+        /// <param name="cancellationToken"></param>
+        /// <returns>null if can't acquire the lock. This is common if somebody else holds it.</returns>
+        Task<IDistributedLock> TryLockAsync(
+            string account,
+            string lockId, 
+            string lockOwnerId,
+            string proposedLeaseId,
+            TimeSpan lockPeriod,  
+            CancellationToken cancellationToken);
+
+        /// <summary>
+        /// Called by the client to renew the lease. The timing internals here are determined by <see cref="SingletonConfiguration"/>
+        /// </summary>
+        /// <param name="lockHandle"></param>
+        /// <param name="cancellationToken"></param>
+        /// <returns>A hint used for determining the next time delay in calling Renew. 
```

Comment:
Can you simplify the contract here? E.g. return true if the lease was renewed successfully, false otherwise? Having a bool return that indicates a delay hint is strange.